### PR TITLE
Adding Protein Structure Prediction related functionality

### PR DESCRIPTION
**Pull request by YoelShoshan**

### CODE REVIEW

File: fusedrug/data/protein/structure/structure_io.py [hunk 1]
```diff
@@ -0,0 +1,747 @@
...
+    """
+    filename = get_pdb_native_full_name(filename)
+    text = read_file_raw_string(filename)
+    protein = protein_utils.from_pdb_string(text, chain_id=chain_id)
+
+    protein.aasequence_str = aa_sequence_from_aa_integers(protein.aatype)
+
+    return protein
+    
+    # with open(filename,'rt') as f:
```
Comment (alex-golts) [inline fusedrug/data/protein/structure/structure_io.py:295]: remove

File: fusedrug/data/protein/structure/structure_io.py [hunk 2]
```diff
@@ -0,0 +1,747 @@
...
+
+    #mmcif_string = read_file_raw_string(reference_mmcif_file)
+    parser = MMCIFParser(QUIET=quiet)
+    if reference_mmcif_file.endswith('.gz'):
+        use_open = gzip.open
+    else:
+        use_open = open
+    with use_open(reference_mmcif_file, 'rt') as fh:
+        ref_structure = parser.get_structure(pdb_id, fh)
+    #parser = MMCIFParser(QUIET=False)
```
Comment (alex-golts) [inline fusedrug/data/protein/structure/structure_io.py:612]: remove

File: fusedrug/data/protein/structure/structure_io.py [hunk 3]
```diff
@@ -0,0 +1,747 @@
...
+
+def read_file_raw_string(filename:str) -> str:
+    use_open = open
+    if filename.endswith('.gz'):
+        use_open = gzip.open
+    with use_open(filename, 'rt') as f:
+        loaded = f.read()
+    return loaded
+
+def save_pdb_file(
```
Comment (alex-golts) [inline fusedrug/data/protein/structure/structure_io.py:399]: can you point to url that explains these terms like atom14, b_factors...

**Review by alex-golts — CHANGES_REQUESTED** [review]
Looks good mostly. See some small comments inline.
Also, unless I missed it, shouldn't there be some code that uses all these test data files. are you planning to add tests/examples?

**Comment by YoelShoshan** [comment]
> Looks good mostly. See some small comments inline. Also, unless I missed it, shouldn't there be some code that uses all these test data files? are you planning to add tests/examples?

Thanks!
Currently most (or possibly all) of the code that uses the test data is found in the jupyter notebooks, we should definitely add more test, I agree!

**Review by alex-golts — APPROVED** [review]

**Review by mosheraboh — APPROVED** [review]
Thanks Yoel.
I found some of those functions very useful when I implemented my dataset!
Please make sure we add the new dependencies to the requirements file.
(When I've tried it, I saw that openfold also missed a few - so I suggest to recreate env and to check it) 
Why did you add the examples from the data to fusedrug?
Is there a demo or test in fusedrug that use it?

File: fusedrug/data/protein/structure/sabdab.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    Or you can manually reach it by browsing to: https://opig.stats.ox.ac.uk/webapps/newsabdab/sabdab/
+        click on "Search Structures"
+        -> "Get All Structures" -> "Get All Structures" -> "Downloads" -> click on "summary file" in the sentence: 
+            "Download the summary file for the structures selected by this search. See help for more details on file formats."
+        
+    """
+    if path is None:
+        if 'SABDAB_DIR' not in os.environ:
+            raise Exception('Could not find env. var SABDAB_DIR')
+        path = os.path.join(os.environ['SABDAB_DIR'],'sabdab_summary_all.tsv')
```
[the remark this answers is not in the document]
Comment (YoelShoshan) [reply fusedrug/data/protein/structure/sabdab.py:19]: Do you mean a relative filename? because the function already can get a full path

**Comment by YoelShoshan** [comment]
> tests

I think that I mostly use the test data in example jupyter notebooks, I'll check

**Comment by mosheraboh** [comment]
> > tests
> 
> I think that I mostly use the test data in example jupyter notebooks, I'll check

If the notebook is not in fusedrug. I suggest moving it to the notebook repo.

File: fusedrug/data/protein/structure/sabdab.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mosheraboh) [reply fusedrug/data/protein/structure/sabdab.py:19]: I mean that the name of the summary table might be different (for example, the table of a summary table with affinity only has a different name)

**Pull request merged**